### PR TITLE
extra_axis_parameters and extra_tikzpicture_parameters

### PR DESCRIPTION
**Pull request by danielhkl**

### CODE REVIEW

**Comment by nschloe** [comment]
Thanks for the PR. Two questions:
What's up with the removed
```python
numpy.flipud(img_array)
```
and why the change in the test phash?

**Comment by danielhkl** [comment]
By working on a unit test for the flipped picture I noticed another bug, which happens when you try to plot an rgb picture. It was also flipped in the pdf file, because PIL.Image.save also don't consider the origin parameter. Some changes I made caused unit test fails, so I corrected the test or the phash. 
Meanwhile, I made a unit test for the flipped picture for a monochrome and a rgb picture. Actually, I just improved the existing unit test 'test_image_plot'. I also added that in case of an unit test fail pdf files are saved in the test directory. Perhaps, I'd better close this PR and start a new tidied up one with all changes till now and some more comments. What do you think ?

File: matplotlib2tikz/image.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def draw_image(data, obj):
         assert len(dims) == 3 and dims[2] in [3, 4]
         # convert to PIL image
         image = PIL.Image.fromarray(img_array)
-        image.save(filename)
+        image.save(filename, origin=obj.origin)
```
Comment (danielhkl) [inline matplotlib2tikz/image.py]: Perhaps, in future it will take the origin parameter into account.

File: test/test_image_plot.py [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
-    from matplotlib import rcParams
-    from matplotlib import pyplot as pp
-    import os
-    try:
-        from PIL import Image
-    except ImportError:
-        raise RuntimeError('PIL must be installed to run this example')
+#the picture 'lena.png' with origin='lower' is flipped upside-down. So it has to be upside-down in the pdf-file as well.
+
+def plot1(): #test for monochrome picture
```
Comment (danielhkl) [inline test/test_image_plot.py]: two tests for rgb and monochrome picture. Now we are able to supervise changes in matplotlib.imsave and PIL.image.save. Hopefully, sometimes the origin parameter will be considered, so we can get rid of the numpy.flipud in matplotlib2tikz.image

**Comment by nschloe** [comment]
I just checked and found that `set()` is bad as a default argument, just like `[]` is:
```python
def f(value, s=set()):
    s.add(value)
    return s

print f('a')
print f('b')
```
Best use `None` and then `if bla is None: # ...`.

**Pull request merged**